### ISSUE TITLE
401 Unauthorized for URLs containing a username

### ISSUE THREAD

**Issue by ThomasJejkal** [lookfirst/sardine#251]
Hello,

I'm using sardine 5.7 to access the simple WebdavServlet provided by the Apache Tomcat server.
The servlet is configured to apply security measures to all HTTP methods. 
I've tried to access a resource as follows: 

`SardineImpl impl = new SardineImpl("webdav", "webdav");`
`List<DavResource> = impl.list("http://webdav@localhost:8080/webdav/resource1/");`

This results in the following exception: 

> Exception in thread "main" com.github.sardine.impl.SardineException: Unexpected response (401 Unauthorized)
	at com.github.sardine.impl.handler.ValidatingResponseHandler.validateResponse(ValidatingResponseHandler.java:48)
	at com.github.sardine.impl.handler.MultiStatusResponseHandler.handleResponse(MultiStatusResponseHandler.java:40)
	at com.github.sardine.impl.handler.MultiStatusResponseHandler.handleResponse(MultiStatusResponseHandler.java:35)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:222)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at com.github.sardine.impl.SardineImpl.execute(SardineImpl.java:1008)
	at com.github.sardine.impl.SardineImpl.propfind(SardineImpl.java:446)
	at com.github.sardine.impl.SardineImpl.list(SardineImpl.java:395)
	at com.github.sardine.impl.SardineImpl.list(SardineImpl.java:385)
	at com.github.sardine.impl.SardineImpl.list(SardineImpl.java:379)

To me it seems that 'propfind' is not using a proper authorization header which results in 3 possibilities to workaround this issue: 

1. Disable security for propfind
2. Provide also the password in the URL
3. Provide no user information in the URL 

Currently, I've chosen option 1 as I cannot influence whether the URL contains user information or not. 
However, shouldn't this be handled by sardine itself by providing proper header information for 'propfind'? 

Thank you for your help.

Cheers,
Thomas.

**Comment by ThomasJejkal** [comment]
After some additional reading and taking a closer look at the source code I have to admit that this is no issue of sardine but a lack of knowledge regarding HTTP protocol handling on my side.

Sorry for opening this issue.

Cheers,
Thomas.

**Comment by lookfirst** [comment]
Just curious, was it this?

https://github.com/lookfirst/sardine/wiki/UsageGuide#preemptive-authentication

**Comment by ThomasJejkal** [comment]
No, I don't think so. Preemptive authentication was something I've tried shortly after opening the issue but it didn't change anything. I've also figured out that with providing username@ not only 'propfind' fails but every secured HTTP method. 

I think the point is that the userinfo potion of the URL is not transferred to the server separately (in the URL) but is 'moved' to the header where it seems to overwrite the authentication set by sardine at instantiation time. I've read about this behaviour in connection with web browser but why should the HTTPClient implementations of Apache behave different? 

It seems that I have to find a way to cleanup URLs from the userinfo before using them. But first I check the HTTPClient library if there is a way to change the behaviour described above.
